### PR TITLE
Filter Accounts by Missing Password Policies

### PR DESCRIPTION
**Pull request by Sutto**
Requesting a non-existant password policy causes an exception when used with the existing `password-policy` - but serves as a good check "do they have a password policy at all?

The existing `password-policy` filter is based on the `ValueFilter` - and so I can't easily add an extra flag to is (as far as I know?). I could modify the data it returns to have a single value (configured or the like) not present in the actual response, and check that value - but that risks breaking backwards compatibility by catching the edge case (but unlikely).

This adds a new, simple filter that just checks for the presence of a password policy and is intended to be easily combinable . Tests included.

Please note: This may hit the object twice in the process of getting it, so may be worth also adding the password policy on to the object given we're fetching it?

### CODE REVIEW

File: c7n/resources/account.py [hunk 1]
```diff
@@ -293,6 +295,46 @@ def process(self, resources, event=None):
             return resources
         return []
 
+@filters.register('has-password-policy')
+class HasAccountPasswordPolicy(Filter):
+  """
```
Comment (kapilt) [inline c7n/resources/account.py]: four legs good, two legs bad... err i mean spaces. re style.

**Comment by kapilt** [comment]
so i think we should be doing this try except for get account password policy in the password-policy filter, effectively if there is no password-policy we can just default it to the default values of those policy attributes if no explicit password policy is set (per https://docs.aws.amazon.com/IAM/latest/APIReference/API_UpdateAccountPasswordPolicy.html) or alternatively to {}. afaics this would remove the need entirely for the has-password-policy as a separate filter, since it would allow verifying actual contents of the policy irrespective of mere existence (ie strong verification). . ie a password policy with a password length of 1, doesn't really say much if just verifying existence vs  actually checking the min length (for the record min length that can be set is 6). ie. do you really want to merely check existence of custom, or verify contents.

**Comment by Sutto** [comment]
I think there exists value in being able to exist "Have I set the default policy" vs. "do I have a custom policy" (especially in combined / and'ed filters) - but I think doing so using the existing filter makes more sense.

I'll rework this PR and send it through - think it's worth adding a flag denoting if it has been set / customised, in addition to the default maybe? That may make more sense once I add more detail.

**Comment by kapilt** [comment]
sounds good, an option on the extant for `policy-present` as a boolean or maybe custom-policy, thanks.

**Comment by Sutto** [comment]
Sounds good - will fix it shortly

**Comment by Sutto** [comment]
How does this look? We add an extra field `PasswordPolicyConfigured` to the object which signifies it's configured? we match case etc to keep in line with the existing structure of the password policy object.
